File: medplum-java/medplum-model/src/main/java/com/fhir/Organization.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableOrganization.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Organization {
  Optional<code> language();

  Optional<List<CodeableConcept>> type();

  Optional<uri> implicitRules();

  String resourceType();

  Optional<Reference> partOf();

  Optional<List<Identifier>> identifier();

  Optional<List<ContactPoint>> telecom();

  Optional<Boolean> active();

  Optional<List<Address>> address();

  Optional<List<Organization_Contact>> contact();

  Optional<Meta> meta();

  Optional<List<Extension>> extension();

  Optional<List<ResourceList>> contained();

  Optional<id> id();

  Optional<String> name();

  Optional<List<Reference>> endpoint();

  Optional<List<Extension>> modifierExtension();

  Optional<Narrative> text();

  Optional<List<String>> alias();

  static ImmutableOrganization.ResourceTypeBuildStage builder() {
    return ImmutableOrganization.builder();
  }
}

File: medplum-typescript/src/model/fhir/Organization.ts
import { Address } from "./Address";
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { ContactPoint } from "./ContactPoint";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Organization_Contact } from "./Organization_Contact";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface Organization {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    active: boolean | null | undefined;
    type: CodeableConcept[] | null | undefined;
    name: string | null | undefined;
    alias: string[] | null | undefined;
    telecom: ContactPoint[] | null | undefined;
    address: Address[] | null | undefined;
    partOf: Reference | null | undefined;
    contact: Organization_Contact[] | null | undefined;
    endpoint: Reference[] | null | undefined;
}
